packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -273,6 +275,11 @@ export class Outbox {
 		// Let the PendingStateManager know that a message was submitted.
 		// In future, need to shift toward keeping batch as a whole!
 		for (const message of batch) {
+			if (message.deserializedContent.type === ContainerMessageType.IdAllocation) {
+				message.deserializedContent.contents.stashedState =
```

Comment:
Adding the stashedState to the contents is very hacky. That may be fine since you plan on updating the stashed state logic but the logic leaks to 3-4 different code blocks across files making it more problematic.
You should try to confine the changes to couple of places only.
One suggestion is to add this to the `localOpMetadata` when creating the message. Then when creating stashed state (getLocalMessage) copy it from `localOpMetadata` to `contents.stashedState`. That code already has special handling for `localOpMetadata` so it's not too bad. Plus, you don't have to worry about deleting stashedState anywhere making the logic cleaner.

Reply:
I've updated the stashed ops implementation to use `localOpMetadata` and to stick the metadata into `contents.stashedState` when `getLocalState()` is called. We still have to delete the `stashedState` in `reSubmit` so that it doesn't go over the wire, but this is definitely an improvement as we no longer leak  to the outbox.